src/scripts/main.js

```diff
@@ -1,7 +1,338 @@
+    cell.textContent = null;
+    row.appendChild(cell);
+  }
+
+  table.appendChild(row);
+}
+
+game.appendChild(table);
+
+const cells = document.querySelectorAll('.field-cell');
+const cellsArray = Array.from(cells);
+
+const generate = () => {
+  const emptyCells = cellsArray.filter((e) => !e.textContent);
+  const numEmptyCells = emptyCells.length;
+  const randomIndex = Math.floor(Math.random() * numEmptyCells);
+  const randomEmptyCell = emptyCells[randomIndex];
+  const randomNumber = Math.random() < 0.9 ? 2 : 4;
+
+  randomEmptyCell.textContent = randomNumber;
+
+  if (!checkAvailableMoves()) {
+    gameOver();
+  }
+
+  addColours();
+};
+
+const keyboardHandler = keyboard;
+
+const starting = () => {
+  score = 0;
+  gameScore.textContent = score;
+  start.textContent = 'Reset';
+  start.classList.add('restart');
+  messageLose.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  messageStart.classList.add('hidden');
+  document.addEventListener('keydown', keyboardHandler);
+
+  generate();
+  generate();
+};
+
+function keyboard(event) {
+  if (event.key === 'ArrowUp') {
+    keyUp();
+    console.log('Натиснута клавіша "Вгору"');
+  } else if (event.key === 'ArrowDown') {
+    keyDown();
+    console.log('Натиснута клавіша "Вниз"');
+  } else if (event.key === 'ArrowLeft') {
+    keyLeft();
+    console.log('Натиснута клавіша "Вліво"');
+  } else if (event.key === 'ArrowRight') {
+    keyRight();
+    console.log('Натиснута клавіша "Вправо"');
+  }
+}
+
+starting();
+
+start.addEventListener('click', () => {
+  cellsArray.forEach((e) => (e.textContent = null));
+  starting();
+});
+
+function checkAvailableMoves() {
+  for (let i = 0; i < cellsArray.length; i++) {
+    if (!cellsArray[i].textContent) {
+      return true;
+    }
+  }
+
+  for (let i = 0; i < cellsArray.length - width; i++) {
+    if (
+      cellsArray[i].textContent === cellsArray[i + width].textContent &&
+      cellsArray[i].textContent !== ''
+    ) {
+      return true;
+    }
+  }
+
+  for (let i = width; i < cellsArray.length; i++) {
+    if (
+      cellsArray[i].textContent === cellsArray[i - width].textContent &&
+      cellsArray[i].textContent !== ''
+    ) {
+      return true;
+    }
+  }
+
+  for (let i = 0; i < cellsArray.length; i++) {
+    if (i % width !== 0) {
+      if (
+        cellsArray[i].textContent === cellsArray[i - 1].textContent &&
+        cellsArray[i].textContent !== ''
+      ) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+const width = 4;
+
+function moveUp() {
+  for (let i = 0; i < 4; i++) {
+    const index = [i, i + width, i + width * 2, i + width * 3];
+    const column = [index.map((e) => cellsArray[e].textContent)];
+    const filteredColumn = column[0].filter((num) => num);
+    const missing = 4 - filteredColumn.length;
+    const zeros = Array(missing).fill(0);
+    const newColumn = filteredColumn.concat(zeros);
+
+    index.forEach((e, index) => {
```

Comment:
```suggestion
    index.forEach((e, i) => {
```

The same thing here with index in the forEach. if was declared before